system/lib/wasmfs/syscalls.cpp

```diff
@@ -502,22 +502,14 @@ static __wasi_fd_t doOpen(path::ParsedParent parsed,
 
   // If O_TRUNC, truncate the file if possible.
   if (flags & O_TRUNC) {
-    if (child->is<DataFile>()) {
-      if (fileMode & WASMFS_PERM_WRITE) {
-        // TODO: Linux still truncates the file when it's opened only for
-        // reading, so we should too. But until we get proper error reporting
-        // from setSize (which we can then ignore here), the OPFS backend will
-        // crash in this case because it does not support truncating files open
-        // only for reading.
-        if (accessMode != O_RDONLY) {
-          child->cast<DataFile>()->locked().setSize(0);
-        }
-      } else {
-        return -EACCES;
-      }
-    } else {
+    if (!child->is<DataFile>()) {
```

Comment:
Should there be a TODO here for symlinks?

Reply:
No, because symlinks are resolved during path parsing. It's not possible for us to observe a symlink here. We already have an assertion for this on line 469.

Reply:
Got it, thanks!